Sources/AdminPanelProvider/Provider.swift

```diff
@@ -24,29 +24,38 @@ public final class Provider: Vapor.Provider {
         var isStorageEnabled = true
         var fromEmail: String?
         var fromName: String?
+        let fileName = "adminpanel"
 
-        if let config = config["mailgun"] {
-            guard let email = config["fromAddress"]?.string else {
+        if let config = config[fileName] {
+            guard let email = config["email", "fromAddress"]?.string else {
                 throw ConfigError.missing(
                     key: ["fromAddress"],
-                    file: "mailgun",
+                    file: fileName,
                     desiredType: String.self
                 )
             }
 
-            fromEmail = email
-
-            guard let name = config["fromName"]?.string else {
+            guard let name = config["email", "fromName"]?.string else {
                 throw ConfigError.missing(
                     key: ["fromName"],
-                    file: "mailgun",
+                    file: fileName,
                     desiredType: String.self
                 )
             }
 
+            fromEmail = email
             fromName = name
+            panelName = config["name"]?.string ?? panelName
+
+            if let userSkinConfig = config["skin"]?.string {
+                skin = PanelConfig.Skin(rawValue: userSkinConfig) ?? skin
+            }
+
+            if let url = config["baseUrl"]?.string {
+                baseUrl = url
+            }
         } else {
-            print("WARNING: couldn't find `mailgun.json`. Email features will be disabled.")
+            print("WARNING: couldn't find `\(fileName).json`. Email features will be disabled.")
```

Comment:
I think we need to consider the case where the file is present, but the email stuff is not configured.